example/glue/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -149,6 +149,13 @@ MM_CollectorLanguageInterfaceImpl::markingScheme_scanRoots(MM_EnvironmentBase *e
 		_markingScheme->markObject(env, rEntry->rootPtr);
 		rEntry = (RootEntry *)hashTableNextDo(&state);
 	}
+	OMR_VMThread *walkThread;
```

Comment:
I agree. But I tried to avoid changing java core structures to reduce scope and complexity of this commit. Java uses internal J9VMThread fields (allocateObjectSavePrivate[12]) to save objects, and these are not accessible to all languages. 

I propose opening a new issue specifically to extend OMRVMThread (provide two slots for saved objects) and then implement save/restoreObjects() in MM_EnvironmentBase as you suggest below.  We can take a look at pulling this out of MM_RootScanner (for Java) in the context of this (new) issue.